kgforge/specializations/stores/bluebrain_nexus.py

```diff
@@ -303,7 +240,7 @@ def _upload_one(self, path: Path, content_type: str) -> Dict:
     # C[R]UD.
 
     def retrieve(
-            self, id_: str, version: Optional[Union[int, str]], cross_bucket: bool, **params
+            self, id_: str, version: Optional[Union[int, str]], cross_bucket: bool = False, **params
```

Comment:
This is done at forge.retrieve() level

Reply:
The issue is that its superclass has the default value for cross_bucket set, and this one didn't so it led to warnings about the implementation having a different signature from its superclass. If I remove it here, I should remove it from the superclass

Reply:
Which warning and what is throwing that ?

Reply:
The signature of the superclass defines the interface that any implementation should follow: all "promises" made at the superclass level should be kept by the implementations. If we want implementation-specific default values, then there shouldn't be a default value on the superclass: the subclass can have a default value without the superclass having one = specialised version of the superclass signature , but not the other way around = more general subclass than superclass